File: Ejercicio 10/ejercicio10.py
class Vehiculo:
    def __init__(self, placa, modelo):
        self.placa = placa
        self.modelo = modelo

    def __str__(self):
        return f"Placa: {self.placa}, Modelo: {self.modelo}"

class EmpresaTransporte:
    def __init__(self, nombre):
        self.nombre = nombre
        self.flota = []

    
    def registrar_vehiculo(self, vehiculo):
        self.flota.append(vehiculo)


if __name__ == "__main__":
    # Vehículos independientes
    auto1 = Vehiculo("MNO-789", "Van Toyota")
    auto2 = Vehiculo("PQR-001", "Furgón Chevrolet")

    empresa = EmpresaTransporte("Carga Rápida")
    

    empresa.registrar_vehiculo(auto1)
    empresa.registrar_vehiculo(auto2)

  
    print(f"Empresa: {empresa.nombre}")
    for v in empresa.flota:
        print(f"  * {v}")
      

File: Ejercicio 10/ejercicio10.java
import java.util.ArrayList;
import java.util.List;

class Vehiculo {
    private String placa;
    private String modelo;

    public Vehiculo(String placa, String modelo) {
        this.placa = placa;
        this.modelo = modelo;
    }

    public String getInfo() {
        return "Vehículo [Placa: " + placa + ", Modelo: " + modelo + "]";
    }
}

class EmpresaTransporte {
    private String nombre;
    private List<Vehiculo> flota;

    public EmpresaTransporte(String nombre) {
        this.nombre = nombre;
        this.flota = new ArrayList<>();
    }

    
    public void agregarAFlota(Vehiculo v) {
        this.flota.add(v);
    }

    public void mostrarFlota() {
        System.out.println("Flota de la empresa: " + nombre);
        for (Vehiculo v : flota) {
            System.out.println(" - " + v.getInfo());
        }
    }
}

public class Main {
    public static void main(String[] args) {

        Vehiculo v1 = new Vehiculo("XYZ-123", "Bus Mercedes");
        Vehiculo v2 = new Vehiculo("ABC-456", "Camión Volvo");

        
        EmpresaTransporte miEmpresa = new EmpresaTransporte("TransLogística S.A.");

        
        miEmpresa.agregarAFlota(v1);
        miEmpresa.agregarAFlota(v2);

        
        miEmpresa.mostrarFlota();
    }
}
